internal/app/init/pkg/network/network.go

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package network
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv4/client4"
+	"github.com/insomniacslk/dhcp/netboot"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+func Setup(platform string) (err error) {
+	link, err := netlink.LinkByName("lo")
+	if err != nil {
+		return err
+	}
+	if err = netlink.LinkSetUp(link); err != nil {
+		return err
+	}
+
+	ifname := "eth0"
+	if _, err = netboot.IfUp(ifname, 5*time.Second); err != nil {
+		return err
+	}
+	modifiers := []dhcpv4.Modifier{dhcpv4.WithRequestedOptions(dhcpv4.OptionHostName, dhcpv4.OptionClasslessStaticRouteOption, dhcpv4.OptionDNSDomainSearchList, dhcpv4.OptionNTPServers)}
+	var netconf *netboot.NetConf
+	if netconf, err = dhclient4(ifname, modifiers...); err != nil {
+		return err
+	}
+	if err = netboot.ConfigureInterface(ifname, netconf); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func dhclient4(ifname string, modifiers ...dhcpv4.Modifier) (*netboot.NetConf, error) {
+	attempts := 10
+	client := client4.NewClient()
+	var (
+		conv []*dhcpv4.DHCPv4
+		err  error
+	)
+	for attempt := 0; attempt < attempts; attempt++ {
+		log.Printf("requesting DHCP lease: attempt %d of %d", attempt+1, attempts)
+		conv, err = client.Exchange(ifname, modifiers...)
+		if err != nil && attempt < attempts {
+			log.Printf("failed to request DHCP lease: %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		break
+	}
+
+	for _, m := range conv {
+		log.Println(m.Summary())
+		if m.OpCode == dhcpv4.OpcodeBootReply && m.MessageType() == dhcpv4.MessageTypeOffer {
+			if m.YourIPAddr != nil {
+				log.Printf("using IP address %s", m.YourIPAddr.String())
+			}
+
+			if m.HostName() != "" {
+				log.Printf("using offered hostname: %s", m.HostName())
+				if err = unix.Sethostname([]byte(m.HostName())); err != nil {
+					return nil, err
+				}
+			} else {
```

Comment:
So the reason for the hard requirement is that we need it for `kubeadm`. Right now we don't support manual configuration of node network settings otherwise I would agree on the condition that we check if the user has supplied one.